tests/unit/synapseclient/extensions/schema_files/data_models/example.model.csv

```diff
@@ -1,84 +1,92 @@
+Check Match None values,,,,,TRUE,DataProperty,,,matchNone MockComponent.checkMatchNonevalues value error,,,,
+Check Recommended,,,,,FALSE,DataProperty,,,recommended,,,,
+Check Ages,,,,,TRUE,DataProperty,,,protectAges,,,,
+Check Unique,,,,,TRUE,DataProperty,,,unique error,,,,
+Check Range,,,,,TRUE,DataProperty,,,inRange 50 100 error,,,,
+Check Date,,,,,TRUE,DataProperty,,,date,string,date,,
+Check NA,,,,,TRUE,DataProperty,,,int::IsNA,,,,
+MockRDB,,,"Component, MockRDB_id, SourceManifest",,FALSE,DataType,,,,,,,
+MockRDB_id,,,,,TRUE,DataProperty,,,int,,,,
+SourceManifest,,,,,TRUE,DataProperty,,,,,,,
+MockFilename,,,"Component, Filename",,FALSE,DataType,,,,,,,
+JSONSchemaComponent,Component to hold attributes for testing JSON Schemas,,"Component, No Rules, No Rules Not Required, String, String Not Required, Enum, Enum Not Required, Date, URL, InRange, Regex, List, List Not Required, List Enum, List Enum Not Required, List Boolean, List, Integer, List InRange",,FALSE,DataType,,,,,,,
+No Rules,,,,,TRUE,DataProperty,,,,,,,
+No Rules Not Required,,,,,FALSE,DataProperty,,,,,,,
+String,,,,,TRUE,DataProperty,,,,string,,,
+String Not Required,,,,,FALSE,DataProperty,,,,string,,,
+Enum,,"ab, cd, ef, gh",,,TRUE,DataProperty,,,,string,,,
+Enum Not Required,,"ab, cd, ef, gh",,,FALSE,DataProperty,,,,string,,,
+Date,,,,,TRUE,DataProperty,,,date,string,date,,
+URL,,,,,TRUE,DataProperty,,,url,string,uri,,
+InRange,,,,,TRUE,DataProperty,,,inRange 50 100,number,,,
+Regex,,,,,TRUE,DataProperty,,,regex search [a-f],string,,,
+List,,,,,TRUE,DataProperty,,,,string_list,,,
+List Not Required,,,,,FALSE,DataProperty,,,,string_list,,,
+List Enum,,"ab, cd, ef, gh",,,TRUE,DataProperty,,,,string_list,,,
+List Enum Not Required,,"ab, cd, ef, gh",,,FALSE,DataProperty,,,,string_list,,,
+List Boolean,,,,,TRUE,DataProperty,,,,boolean_list,,,
+List Integer,,,,,TRUE,DataProperty,,,,integer_list,,,
+List InRange,,,,,TRUE,DataProperty,,,inRange 50 100,integer_list,,,
+TypeDefinitionComponent,Component to check type specification,,"Component, String type, String type caps, Int type, Int type caps, Num type, Num type caps, Nan type, Missing type, Boolean type, Boolean type caps",,FALSE,DataType,,,,,,,
+String type,,,,,TRUE,DataProperty,,,,string,,,
+String type caps,,,,,TRUE,DataProperty,,,,STRING,,,
+Int type,,,,,TRUE,DataProperty,,,,integer,,,
+Int type caps,,,,,TRUE,DataProperty,,,,INTEGER,,,
+Num type,,,,,TRUE,DataProperty,,,,number,,,
+Num type caps,,,,,TRUE,DataProperty,,,,NUMBER,,,
+Nan type,,,,,TRUE,DataProperty,,,,nan,,,
+Missing type,,,,,TRUE,DataProperty,,,,,,,
+Boolean type,,,,,TRUE,DataProperty,,,,boolean,,,
+Boolean type caps,,,,,TRUE,DataProperty,,,,BOOLEAN,,,
+RangeComponent,Component to ensure maximum and minimum can be set correctly,,"Component, Maximum Integer, Minimum Integer, Maximum Float, Minimum Float, Maximum Minimum, Maximum Minimum Integer List",,FALSE,DataType,,,,,,,
```

Comment:
I created this `RangeComponent` to make sure all the following cases can be tested: 
Tests numeric constraint handling with different column types:

* Maximum Integer - Integer type with maximum constraint only (100)
* Minimum Integer - Integer type with minimum constraint only (10)
* Maximum Float - Number type with maximum float value (100.5)
* Minimum Float - Number type with minimum float value (10.8)
* Maximum Minimum - Integer type with both constraints (min: 10, max: 100)
* Maximum Minimum Integer List - Integer_list type with both constraints (min: 10, max: 100)
* Maximum Minimum Validation Rule - Tests precedence: explicit columns (min: 10, max: 200) override deprecated inRange rule (50-100)